cpu/esp32/vendor/esp-idf/driver/Kconfig

```
@@ -0,0 +1,11 @@
+# Copyright (c) 2021 HAW Hamburg
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+#
+
+config MODULE_ESP_IDF_DRIVER
+    bool
+    depends on TEST_KCONFIG
+    depends on HAS_ARCH_ESP
```

Here an for all others in subdirectories of `cpu/esp32/vendor/esp-idf`
```suggestion
    depends on HAS_ARCH_ESP32
```
The vendor code is specific for ESP32.

---

Now it depends on `MODULE_ESP_IDF`, which is already depending on the arch feature